### ISSUE TITLE
instances don't work with setuid and hidepid on /proc

### ISSUE THREAD

**Issue by nlvw** [apptainer/apptainer#850]
When trying to use the Apptainer Instance feature it will fail if you are using setuid and have hidepid set on /proc.  This failure happens even if you allow the current user to bypass hidepid.

```
$ cat /etc/fstab | grep /proc
proc /proc proc defaults,hidepid=2,gid=1500 0 0

$  getent group 1500
monitoring:x:1500:polkitd,zabbix,ara,semaphore

$ id
uid=1002(ara) gid=1002(ara) groups=1002(ara),1500(monitoring) context=unconfined_u:unconfined_r:unconfined_t:s0-s0:c0.c1023

$ /usr/bin/apptainer instance start --pid-file "/opt/ara/ara_data/ara.pid" "/opt/ara/ansible-ict-systems-monolithic/.cache/container.sif" ara "ara"
FATAL:   hidepid option set on /proc mount, require 'hidepid=0' to start instance with setuid workflow

```

`--userns` can be used to bypass this issue but if you have a service that requires using the ssh client inside the container then you have to use the setuid workflow instead of userns or SSH will get mad about things not being correctly owned by root.

### Version of Apptainer

```console
$ apptainer --version
apptainer version 1.1.2-1.el8
```

### What OS/distro are you running?

```console
$ cat /etc/os-release
NAME="Red Hat Enterprise Linux"
VERSION="8.6 (Ootpa)"
ID="rhel"
ID_LIKE="fedora"
VERSION_ID="8.6"
PLATFORM_ID="platform:el8"
PRETTY_NAME="Red Hat Enterprise Linux 8.6 (Ootpa)"
ANSI_COLOR="0;31"
CPE_NAME="cpe:/o:redhat:enterprise_linux:8::baseos"
HOME_URL="https://www.redhat.com/"
DOCUMENTATION_URL="https://access.redhat.com/documentation/red_hat_enterprise_linux/8/"
BUG_REPORT_URL="https://bugzilla.redhat.com/"

REDHAT_BUGZILLA_PRODUCT="Red Hat Enterprise Linux 8"
REDHAT_BUGZILLA_PRODUCT_VERSION=8.6
REDHAT_SUPPORT_PRODUCT="Red Hat Enterprise Linux"
REDHAT_SUPPORT_PRODUCT_VERSION="8.6"
```

### How did you install Apptainer?

EPEL 8 Repositories

**Comment by DrDaveD** [comment]
Can you try 1.1.3, now in EPEL testing, without suid mode?  It has a fix that allowed ssh to run inside a regular container.  I didn't try it with an instance.

**Comment by nlvw** [comment]
We don't have EPEL testing setup in our Satellite instance but I'll test it as soon as it's in the regular repo.  

That being said unless the project is retiring setuid entirely then improving compatibility with hidepid on /proc would be a nice to have enhancement.  Whether that is supporting hidepid when the user is in the allow group (improve check before throwing the fetal error) or supporting setuid when you can only see your own processes.

**Comment by DrDaveD** [comment]
@nlvw have you been able to try it without setuid?

Did this used to work with setuid on older singularity/apptainer versions?

**Comment by nlvw** [comment]
I haven't.  For my use case the setuid is important for root file ownership to show correctly. 

I looked at the mention of the userns workaround but I couldn't make sense of what I actually need to do to get it working.  What flags to set and if the users uid still works correctly when using said flag (I don't want to be root inside container)

This is my first time using instances so I don't know about previous versions.

For now I've disabled hidepid on that particular server.
